finite-state/shared/src/main/scala/fs2/data/pattern/DecisionTree.scala

```diff
@@ -39,7 +37,10 @@ sealed trait DecisionTree[Tag, Out] {
                   }
               }
             case None =>
-              None
+              fallBack match {
+                case Some(tree) => loop(tree)
+                case None       => None
+              }
```

Comment:
Isn't that effectively a `flatMap`?

```suggestion
              fallBack.flatMap(loop)
```

Reply:
Yes it is, but `@tailrec` optimization is here in my implementation :sweat_smile:

Reply:
OK, I let that argument count 😄